node/resolvers/createReturnRequest.ts

```diff
@@ -69,12 +70,12 @@ export const createReturnRequest = async (
 
   const itemsToReturn = createItemsToReturn(items, orderItems)
 
-  // TODO: Apply the flag to return proportional shipping from admin
-  const shippingAmount = totals.find(({ id }) => id === 'Shipping')?.value ?? 0
+  const settings = await appSettings.get(SETTINGS_PATH, true)
 
   const refundableAmountTotals = createRefundableTotals(
     itemsToReturn,
-    shippingAmount
+    totals,
+    settings?.options?.enableProportionalShippingValue as boolean
```

Comment:
This value might not be a `boolean` and can be `undefined`. In this case, it's better to not cast the type.